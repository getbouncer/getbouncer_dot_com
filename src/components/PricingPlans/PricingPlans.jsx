import React from 'react'
import PropTypes from 'prop-types'
import Container from '~globals/Container'
import CtaButton from '~components/CtaButton'
import { Smartphone } from '~components/Media'
import {
  Headline,
  Wrapper,
  Card,
  CardFooter,
  CardFooterText,
  CardsContainer,
  CardText,
  CardPrice,
  CtaWrapper,
  Subtitle,
  SubtitleLink,
} from './styles'

const PricingPlans = ({ data }) => (
  <Wrapper>
    <Container>
      <Headline>{data.headline.text}</Headline>
      <CardsContainer>
        {data.body.map(card => (
          <Card key={card.primary.tier_name}>
            <h3>{card.primary.tier_name}</h3>
            <CardText>{card.primary.tier_description}</CardText>
            <CardPrice>{card.primary.tier_price}</CardPrice>
            <CtaWrapper>
              <CtaButton text={card.primary.tier_cta} to="/" />
            </CtaWrapper>
            <Smartphone>
              <CardFooter>
                {card.items.map(item => (
                  <CardFooterText key={item.tier_feature}>{item.tier_feature}</CardFooterText>
                ))}
              </CardFooter>
            </Smartphone>
          </Card>
        ))}
      </CardsContainer>
      <Subtitle>
        {data.text.text}
        &nbsp;
        <SubtitleLink to="/contact">Letâ€™s chat.</SubtitleLink>
      </Subtitle>
    </Container>
  </Wrapper>
)

PricingPlans.propTypes = {
  data: PropTypes.shape().isRequired,
}

export default PricingPlans
