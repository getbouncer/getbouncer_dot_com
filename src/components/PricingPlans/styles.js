import styled from '@emotion/styled'
import { Link } from 'gatsby'
import { breakpoints, theme } from '~styles'
import checkedIcon from '~images/checked.svg'
import checkedIconBlue from '~images/checked-blue.svg'
import gradientBg from '~images/pricing-background.png'

export const Wrapper = styled.section`
  background-image: url(${gradientBg});
  background-repeat: no-repeat;
  background-size: 120% 100%;
  background-position: -50px 0;
  padding: 100px 0;

  @media ${breakpoints.smartphone} {
    padding: 50px 0 5px;
  }
`

export const Headline = styled.h1`
  text-align: center;
  margin-bottom: 50px;
`

export const CardsContainer = styled.div`
  align-items: center;
  display: flex;
  justify-content: center;

  @media ${breakpoints.smartphone} {
    display: block;
  }
`

export const Card = styled.div`
  border-radius: 10px;
  box-shadow: -4px 15px 40px 10px rgba(85, 105, 211, 0.36);
  padding: 34px 24px 65px;
  text-align: center;
  width: 336px;

  @media ${breakpoints.smartphone} {
    margin-bottom: 48px;
    width: 100%;
    padding: 22px 15px 32px;
  }

  &:first-of-type {
    background: #fff;
    z-index: 0;

    span {
      @media ${breakpoints.smartphone} {
        margin: 40px 0;
      }
    }
  }
  &:nth-of-type(2) {
    background: #e9ecfe;
    z-index: 10;
  }
  &:nth-of-type(3) {
    background: #5569cf;
    color: #fff;
    padding: 45px 24px 65px;
    z-index: 20;

    a {
      background-color: ${theme.colors.white};
      color: ${theme.colors.blue};
    }

    div:last-of-type p::before {
      background-color: ${theme.colors.white};
      background-image: url(${checkedIconBlue});
    }
  }
`

export const CardText = styled.p`
  font-size: 20px;
  display: block;
  margin: 25px auto 0;

  @media ${breakpoints.smartphone} {
    margin: 18px auto 0;
  }
`

export const CardPrice = styled.span`
  display: block;
  font-family: 'Halyard Display';
  font-size: 52.5px;
  font-weight: 500;
  margin: 70px 0;

  @media ${breakpoints.smartphone} {
    margin: 20px 0 40px;
  }
`

export const CtaWrapper = styled.div`
  text-align: center;
`

export const Subtitle = styled.p`
  display: block;
  margin: 45px auto 0;
  text-align: center;
  max-width: 587px;
`

export const SubtitleLink = styled(Link)`
  color: ${theme.colors.blue};
`

export const CardFooter = styled.div`
  border-top: 1px solid ${theme.colors.lightBlue};
  margin-top: 50px;
  padding: 40px 32px 0;
  width: 100%;
`

export const CardFooterText = styled.p`
  display: block;
  margin-left: 36px;
  position: relative;
  text-align: left;

  &::before {
    background-color: ${theme.colors.blue};
    background-image: url(${checkedIcon});
    background-position: 50%;
    background-repeat: no-repeat;
    background-size: 12px;
    border-radius: 100%;
    content: '';
    height: 22px;
    left: -36px;
    position: absolute;
    width: 22px;
  }

  &:not(:last-of-type) {
    margin-bottom: 24px;
  }
`
