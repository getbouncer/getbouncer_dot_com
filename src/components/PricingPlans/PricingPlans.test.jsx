import React from 'react'
import renderer from 'react-test-renderer'
import PricingPlans from './PricingPlans'

/* eslint-disable no-undef */

describe('PricingPlans', () => {
  it('renders correctly', () => {
    const data = {
      headline: {
        text: 'headline',
      },
      comparison_headline: {
        text: 'comparison headline',
      },
      text: {
        text: 'text',
      },
      body: [
        {
          id: 'id 1',
          primary: {
            tier_link_internal: true,
            tier_link: 'https://www.example.com/tier_1',
            tier_cta: 'tier 1 cta',
            tier_description: 'tier 1 description',
            tier_name: 'tier 1',
            tier_short_description: 'tier 1 short description',
            tier_price: 'tier 1 price',
          },
          items: [
            {
              tier_feature: 'tier 1 feature',
            },
            {
              tier_feature: 'tier 1 feature',
            },
          ],
        },
        {
          id: 'id 2',
          primary: {
            tier_link_internal: false,
            tier_link: 'https://www.example.com/tier_2',
            tier_cta: 'tier 2 cta',
            tier_description: 'tier 2 description',
            tier_name: 'tier 2',
            tier_short_description: 'tier 2 short description',
            tier_price: 'tier 2 price',
          },
          items: [
            {
              tier_feature: 'tier 2 feature',
            },
            {
              tier_feature: 'tier 2 feature',
            },
          ],
        },
      ],
    }
    const tree = renderer
      .create(<PricingPlans data={data} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
