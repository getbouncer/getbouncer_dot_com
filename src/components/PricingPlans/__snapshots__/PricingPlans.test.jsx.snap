// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PricingPlans renders correctly 1`] = `
<section
  className="css-161o3a0"
>
  <div
    className="css-1fywgw4"
  >
    <h1
      className="css-hq5whk"
    >
      headline
    </h1>
    <div
      className="css-d71ng7"
    >
      <div
        className="css-1yts8x9"
      >
        <h3>
          tier 1
        </h3>
        <p
          className="css-a5dgoa"
        >
          tier 1 description
        </p>
        <span
          className="css-1tkq62a"
        >
          tier 1 price
        </span>
        <div
          className="css-1yltisf"
        >
          <a
            className="css-e1lh64"
            href="https://www.example.com/tier_1"
            inverted={false}
          >
            tier 1 cta
          </a>
        </div>
      </div>
      <div
        className="css-1yts8x9"
      >
        <h3>
          tier 2
        </h3>
        <p
          className="css-a5dgoa"
        >
          tier 2 description
        </p>
        <span
          className="css-1tkq62a"
        >
          tier 2 price
        </span>
        <div
          className="css-1yltisf"
        >
          <a
            className="css-e1lh64"
            href="https://www.example.com/tier_2"
          >
            tier 2 cta
          </a>
        </div>
      </div>
    </div>
    <p
      className="css-hbrf8g"
    >
      text
       
      <a
        className="css-13zgsqv"
        href="/contact"
      >
        Let’s chat.
      </a>
    </p>
  </div>
</section>
`;
